File: src/features/typing/data/ts/reverse_nodes_in_k_group.ts
class ListNode {
    val: number;
    next: ListNode | null;
    constructor(val = 0, next: ListNode | null = null) {
        this.val = val;
        this.next = next;
    }
}

function reverseKGroup(head: ListNode | null, k: number): ListNode | null {
    let node: ListNode | null = head;
    for (let i = 0; i < k; i++) {
        if (node === null) {
            return head;
        }
        node = node.next;
    }
    let prev: ListNode | null = null;
    let cur: ListNode | null = head;
    for (let i = 0; i < k; i++) {
        const next: ListNode | null = cur!.next;
        cur!.next = prev;
        prev = cur;
        cur = next;
    }
    head!.next = reverseKGroup(cur, k);
    return prev;
}

File: src/features/typing/data/go/reverse_nodes_in_k_group.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func reverseKGroup(head *ListNode, k int) *ListNode {
	node := head
	for i := 0; i < k; i++ {
		if node == nil {
			return head
		}
		node = node.Next
	}
	var prev *ListNode
	cur := head
	for i := 0; i < k; i++ {
		next := cur.Next
		cur.Next = prev
		prev = cur
		cur = next
	}
	head.Next = reverseKGroup(cur, k)
	return prev
}
